db/backup_download.go

```diff
@@ -102,6 +103,15 @@ func DownloadBackup(app, addon, backupID string, opts DownloadBackupOpts) error
 		return errgo.Notef(err, "fail to start download")
 	}
 	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		spinner.Stop()
```

Comment:
Sorry I didn't noticed it at the first review, but can you move this line before the `if`, and remove the call to `spinner.STop` on line 115?